types/jest-when/index.d.ts

```diff
@@ -10,21 +10,41 @@
 
 /// <reference types="jest" />
 
-export interface WhenMock<T = any, Y extends any[] = any>
-  extends jest.MockInstance<T, Y> {
-  calledWith(...matchers: Y): this;
-  expectCalledWith(...matchers: Y): this;
-  mockReturnValue(value: T): this;
-  mockReturnValueOnce(value: T): this;
-  mockResolvedValue(value: jest.ResolvedValue<T>): this;
-  mockResolvedValueOnce(value: jest.ResolvedValue<T>): this;
-  mockRejectedValue(value: jest.RejectedValue<T>): this;
-  mockRejectedValueOnce(value: jest.RejectedValue<T>): this;
-  mockImplementation(fn: (...args: Y) => T): this;
-  mockImplementationOnce(fn?: (...args: Y) => T): this;
+export type ArgumentOrMatcher<ArgTypes extends any[]> = {
+    [Index in keyof ArgTypes]: ArgTypes[Index] | WhenMock<boolean, [ArgTypes[Index]]>;
```

Comment:
Should this be `number` or `keyof ArgTypes & number`?  It doesn't seem intended to accept, e.g., `"length"`.

Reply:
It certainly appears to do the right thing in practice. :smile: